### PR TITLE
Build and test amd64 docker images in a dedicated workflow for quick feedback

### PR DESCRIPTION
**Pull request by mathbunnyru**
The idea behind this is simple - let's build x86 images in a different workflow.
It's enough for most PRs and we can skip long x86+arm builds and merge quicker.

### CODE REVIEW

File: .github/workflows/docker-x86.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      - "r-notebook/**"
+      - "scipy-notebook/**"
+      - "tensorflow-notebook/**"
+
+      - "tagging/**"
+      - "test/**"
+      - "conftest.py"
+      - "Makefile"
+      - "pytest.ini"
+      - "requirements-dev.txt"
```
Comment (consideRatio) [inline .github/workflows/docker-x86.yml:44]: Allows us to run this manually, which is very useful to verify if something has broken due to a PR or due to a change over time.

```suggestion
      - "requirements-dev.txt"
  workflow_dispatch:
```

File: .github/workflows/docker-x86.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
+name: Build and test x86 Docker Images
```
Comment (consideRatio) [inline .github/workflows/docker-x86.yml:1]: Regarding referencing this as x86, when using docker buildx, we refer to x86-64 as amd64 I think. I suggest we stick with the amd64 reference for consistency.

Comment (consideRatio) [inline .github/workflows/docker-x86.yml:1]: Can you add a descriptive comment in the top to clarify the purpose of having this dedicated workflow file?

The point as I understand it, is to run another workflow that will finish a lot quicker, allowing us to conclude if things work or not far quicker, when we don't expect there to be any platform specific breaking change.

File: .github/workflows/docker-x86.yml (hunk 1 above)
Comment (mathbunnyru) [reply .github/workflows/docker-x86.yml:44]: Nice idea!
Maybe it's worth adding to other workflows as well.

File: .github/workflows/docker-x86.yml (hunk 2 above)
Comment (mathbunnyru) [reply .github/workflows/docker-x86.yml:1]: Done

Comment (mathbunnyru) [reply .github/workflows/docker-x86.yml]: Yes, you got it right. Done.

**Comment by mathbunnyru** [comment]
Nice, it takes less than 30 minutes, merging this one.

**Pull request merged**